### ISSUE TITLE
Diff review mode broken due to trusted type policy

### ISSUE THREAD

**Issue by MarcoZehe** [microsoft/vscode#115224]
- VSCode Version: Version: 1.53.0-insider
Commit: 7b0cfdd04ad530a9c8d782c618afd79290be3d64
Date: 2021-01-27T08:59:36.340Z
Electron: 11.2.1
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Darwin arm64 20.3.0
- OS Version: MacOS 11.2

Steps to Reproduce:

1. Open a versioned file and make some changes.
2. Press F7 to jump to the differences.

Result: A warning comes up in the notifications saying "Cannot set property 'tabIndex' of null". And the side bar with the diffs doesn't open. Alternatively, a different warning is displayed saying "UNTRUSTED html usage, default trusted types policy should NEVER be reached".

This is a fairly new regression in the latest Insider builds, haven't seen this in earlier versions.

Does this issue occur when all extensions are disabled?: Yes

CC @isidorn

**Comment by isidorn** [comment]
@MarcoZehe thanks a lot for filling this

DiffReview mode seems to be broken due to our trusted type work. Assigning also to @jrieken and @alexdima since they might have more ideas how to fix.

here's the call stack

```
ERR UNTRUSTED html usage, default trusted types policy should NEVER be reached: Error: UNTRUSTED html usage, default trusted types policy should NEVER be reached
    at createHTML (file:///Users/isidor/Development/vscode/out/vs/code/electron-browser/workbench/workbench.js:59:10)
    at Function._renderSection (file:///Users/isidor/Development/vscode/out/vs/editor/browser/widget/diffReview.js:575:26)
    at DiffReview._render (file:///Users/isidor/Development/vscode/out/vs/editor/browser/widget/diffReview.js:487:28)
    at file:///Users/isidor/Development/vscode/out/vs/editor/browser/widget/diffReview.js:68:22
    at Emitter.fire (file:///Users/isidor/Development/vscode/out/vs/base/common/event.js:479:38)
    at file:///Users/isidor/Development/vscode/out/vs/editor/browser/widget/diffEditorWidget.js:776:43
```

The call to [`insertAdjacentHTML`](https://github.com/microsoft/vscode/blob/096d70ce18efa17e04908094c3052ad0fa1a5ab9/src/vs/editor/browser/widget/diffReview.ts#L737) seems to no longer be allowed.

**Comment by isidorn** [comment]
Using `insertAdjacentText` instead seems to fix the issue. Pushing the fix and you can review @jrieken @alexdima

**Comment by isidorn** [comment]
I did a full text search and we seem to have one more usage of `insertAdjacentHtml` here
https://github.com/microsoft/vscode/blob/096d70ce18efa17e04908094c3052ad0fa1a5ab9/src/vs/editor/browser/view/viewLayer.ts#L517
@alexdima let me know if you would like me to also change that or should I create an issue

**Comment by alexdima** [comment]
@isidorn eb1cf4b0bae2b0f3af9681b29efade1f09f01a08 is not a good change. You cannot take HTML and insert it as text.

**Comment by alexdima** [comment]
![image](https://user-images.githubusercontent.com/5047891/106003572-9ed2ea80-60b2-11eb-84b4-73a0a7380292.png)

**Comment by alexdima** [comment]
I have pushed a revert for now. I can look into adopting Trusted Types here.

**Comment by isidorn** [comment]
That was foolish of me. Thanks for fixing it properly.

**Comment by jrieken** [comment]
Thanks everyone

**Comment by isidorn** [comment]
@alexdima just one more ping regarding my comment from above (so we do not lose it) we are also using `insertAdjacentHtml` [here](https://github.com/microsoft/vscode/blob/096d70ce18efa17e04908094c3052ad0fa1a5ab9/src/vs/editor/browser/view/viewLayer.ts#L517)
I am not sure if that is ok

**Comment by jrieken** [comment]
I feel that my education on trusted types was a total failure. @isidorn if you scroll up a little you'll see that it is created with a policy

**Comment by alexdima** [comment]
Yeah, that usage in `viewLayer` is fine, otherwise the editor would never render. @jrieken just FYI I had to change a `Content-Security-Policy`. Not sure if we document this somewhere for downstream workbench users.

**Comment by jrieken** [comment]
> @jrieken just FYI I had to change a Content-Security-Policy. Not sure if we document this somewhere for downstream workbench users.

Not yet - I first want to get more feedback (like this issue) and then a pass over all current policies.
